lib/Sema/TypeCheckType.cpp

```diff
@@ -2449,7 +2449,6 @@ Type TypeResolver::resolveOpaqueReturnType(TypeRepr *repr,
   if (definingDeclNode->getKind() == Node::Kind::Global)
     definingDeclNode = definingDeclNode->getChild(0);
   ASTBuilder builder(Context);
-  builder.Resolver = resolution.Resolver;
```

Comment:
You might be able to drop the `Resolver` from the `resolution` structure now too. I think I needed to tunnel it through for the hack implementation previously.

Reply:
Ah, good catch.